cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -523,14 +535,18 @@ def with_measurement_key_mapping(self, key_map: Dict[str, str]) -> 'CircuitOpera
                 keys than this operation.
         """
         new_map = {}
-        for k in self.circuit.all_measurement_key_names():
+        for k in self.circuit.all_measurement_key_names() | protocols.control_key_names(
+            self.circuit
+        ):
             k = value.MeasurementKey.parse_serialized(k).name
             k_new = self.measurement_key_map.get(k, k)
             k_new = key_map.get(k_new, k_new)
             if k_new != k:
                 new_map[k] = k_new
         new_op = self.replace(measurement_key_map=new_map)
-        if len(new_op._measurement_key_names_()) != len(self._measurement_key_names_()):
+        if len(new_op._measurement_key_names_()) != len(self._measurement_key_names_()) or len(
```

Comment:
I think we also need a check that this doesn't cause any new conflicts between control keys and measurement keys. Maybe check `len(measurement_keys & control_keys)` is unchanged. I'm *pretty sure* we need all three checks. Write some unit tests that cover each case.